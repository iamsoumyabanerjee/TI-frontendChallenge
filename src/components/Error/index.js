import React from 'react'
import styles from './styles.scss'
const Error = ({ content = 'Error getting the data, please try after some time' }) => (<div className='error'><p>{content}</p></div>)

export default Error