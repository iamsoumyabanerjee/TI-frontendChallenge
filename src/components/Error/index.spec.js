import React from 'react'
import renderer from 'react-test-renderer'
import Error from '.'

describe('#Error', () => {
  it('should render correctly with default text', () => {
    const tree = renderer
      .create(<Error />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with supplied text', () => {
    const props = { content: 'Some content' }
    const tree = renderer
      .create(<Error { ... props} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})