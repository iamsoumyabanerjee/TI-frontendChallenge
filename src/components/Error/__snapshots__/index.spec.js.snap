// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Error should render correctly with default text 1`] = `
<div
  className="error"
>
  <p>
    Error getting the data, please try after some time
  </p>
</div>
`;

exports[`#Error should render correctly with supplied text 1`] = `
<div
  className="error"
>
  <p>
    Some content
  </p>
</div>
`;
